File: TI1.1 SOAP_JAVA_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Java_GR06/src/java/ec/edu/monster/services/ConversorMasa.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.services;

/**
 *
 * @author josue
 */
public class ConversorMasa {
    
    // Kilogramo <-> Gramo
    public double kilogramoAGramo(double kilogramos) {
        return kilogramos * 1000.0;
    }
    
    public double gramoAKilogramo(double gramos) {
        return gramos / 1000.0;
    }
    
    // Kilogramo <-> Libra
    public double kilogramoALibra(double kilogramos) {
        return kilogramos * 2.20462;
    }
    
    public double libraAKilogramo(double libras) {
        return libras / 2.20462;
    }
    
    // Gramo <-> Libra
    public double gramoALibra(double gramos) {
        return kilogramoALibra(gramoAKilogramo(gramos));
    }
    
    public double libraAGramo(double libras) {
        return kilogramoAGramo(libraAKilogramo(libras));
    }
}

File: TI1.4 RESTFUL_DOTNET_SINBDD_GR06/01 SERVIDOR/ConUni_RestFul_Dotnet_GR06/ConUni_RestFul_Dotnet_GR06/Servicios/ConversorMasa.cs
namespace ConUni_RestFul_Dotnet_GR06.Servicios
{
    /// <summary>
    /// Servicio de conversión de masa
    /// </summary>
    public class ConversorMasa
    {
        // Kilogramo <-> Gramo
        public double KilogramoAGramo(double kilogramos)
        {
            return kilogramos * 1000.0;
        }

        public double GramoAKilogramo(double gramos)
        {
            return gramos / 1000.0;
        }

        // Kilogramo <-> Libra
        public double KilogramoALibra(double kilogramos)
        {
            return kilogramos * 2.20462;
        }

        public double LibraAKilogramo(double libras)
        {
            return libras / 2.20462;
        }

        // Gramo <-> Libra
        public double GramoALibra(double gramos)
        {
            return KilogramoALibra(GramoAKilogramo(gramos));
        }

        public double LibraAGramo(double libras)
        {
            return KilogramoAGramo(LibraAKilogramo(libras));
        }
    }
}
